// actions/actionCreators.js
//functions actions creators
//fetch phone to state
import {
    ADD_PHONE_TO_BASKET, CLEAN_BASKET, FETCH_CATEGORIES_LOADING,
    FETCH_PHONE_BY_ID_FAILURE,
    FETCH_PHONE_BY_ID_LOADING, FETCH_PHONE_BY_ID_SUCCESS,
    FETCH_PHONE_FAILURE,
    FETCH_PHONE_LOADING,
    FETCH_PHONE_SUCCESS, LOAD_MORE_PHONE_FAILURE,
    LOAD_MORE_PHONE_LOADING, LOAD_MORE_PHONE_SUCCESS, REMOVE_PHONE_FROM_BASKET, SEARCH_PHONE
} from "actions/actionsTypes";

export const FetchPhoneAC = {type:FETCH_PHONE_LOADING};
export const FetchPhoneScAC =(phones) => ({type:FETCH_PHONE_SUCCESS , payload:phones});
export const FetchPhoneFlAC =(err)=> ({type:FETCH_PHONE_FAILURE , payload:err , error:true});
//load more phones to state
export const LoadMorePhoneAC = {type:LOAD_MORE_PHONE_LOADING};
export const LoadMorePhoneScAC =(phones) => ({type:LOAD_MORE_PHONE_SUCCESS , payload:phones});
export const LoadMorePhoneFlAC =(err)=> ({type:LOAD_MORE_PHONE_FAILURE , payload:err , error:true});
//take phone by id and push it to state
export const FetchPhoneByIDAC = {type:FETCH_PHONE_BY_ID_LOADING};
export const FetchPhoneByIDScAC =(phone) => ({type:FETCH_PHONE_BY_ID_SUCCESS , payload:phone});
export const FetchPhoneByIDFlAC =(err)=> ({type:FETCH_PHONE_BY_ID_FAILURE , payload:err , error:true});
//add phone to basket state
export const AddPhoneToBasketAC = (id)=>({type:ADD_PHONE_TO_BASKET,payload:id})
//give phone by name
export const SeatchPhoneAC = text =>({type:SEARCH_PHONE , payload:text})
//fetch categories
export const FetchCategoriesAC = categories =>({type:FETCH_CATEGORIES_LOADING,payload :categories})
//remove phone from basket
export const RemovePhoneAC = id =>({type:REMOVE_PHONE_FROM_BASKET,payload :id})
//clearBasket
export const ClearBasketAC = {type:CLEAN_BASKET}
