// actions/actionsTypes.js
//action types in const
export  const FETCH_PHONE_LOADING = "FETCH_PHONE_LOADING";
export const FETCH_PHONE_SUCCESS = "FETCH_PHONE_SUCCES";
export const FETCH_PHONE_FAILURE = "FETCH_PHONE_FAIL";
export  const LOAD_MORE_PHONE_LOADING = "LOAD_MORE_PHONE_LOADING";
export const LOAD_MORE_PHONE_SUCCESS = "LOAD_MORE_PHONE_SUCCES";
export const LOAD_MORE_PHONE_FAILURE = "LOAD_MORE_PHONE_FAIL";
export  const FETCH_PHONE_BY_ID_LOADING = "FETCH_PHONE_BY_ID_LOADING";
export const FETCH_PHONE_BY_ID_SUCCESS = "FETCH_PHONE_BY_ID_SUCCES";
export const FETCH_PHONE_BY_ID_FAILURE = "FETCH_PHONE_BY_ID_FAIL";
export const ADD_PHONE_TO_BASKET = "ADD_PHONE_TO_BASKET";
export const SEARCH_PHONE = "SEARCH_PHONE"
export  const FETCH_CATEGORIES_LOADING = "FETCH_CATEGORIES_LOADING";
export  const FETCH_CATEGORIES_SUCCESS = "FETCH_CATEGORIES_SUCCESS";
export  const FETCH_CATEGORIES_FAILURE = "FETCH_CATEGORIES_FAILURE";
export const REMOVE_PHONE_FROM_BASKET ='REMOVE_PHONE_FROM_BASKET';
export const CLEAN_BASKET = 'CLEAN_BASKET'
